### ISSUE TITLE
Http.sys server should support setting up URL groups when attaching to an existing queue

### ISSUE THREAD

**Issue by NGloreous** [dotnet/aspnetcore#40359]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

When using http.sys delegation, there are two processes involved, the delegator (the process creating the DelegationRule to delegate requests it receives) and the receiver (the process the delegator transfers requests to). It's possible that the receiver may restart independently from the delegator. When using the http.sys server as a receiver with the `RequestQueueMode.Create` option, the receiver fails to start up because the delegator still has a handle to the queue. If the `RequestQueueMode.CreateOrAttach` option is used, the receiver start up fine but fails to receive any traffic because there are no URL groups configured in http.sys.

### Describe the solution you'd like

Either add a new option to RequestQueueMode to indicate that URL groups should be configured when attaching or always try to create these when attaching if specified and they don't already exist.

**Comment by Tratcher** [comment]
Could creating the URL group cause a conflict with another process that has added the same prefixes?

**Comment by NGloreous** [comment]
For the same queue or a different one? If for a different one, doesn't that issue already exist today? I don't think http.sys will let you create a URL group if it would collide with an existing one, but I could try it out.

**Comment by Tratcher** [comment]
If you have multiple instances of the same app all running in CreateOrAttach mode with the same prefixes and queue, would they conflict with eachother if they all made the same url group?

**Comment by NGloreous** [comment]
Yes, but clarification on terminology, creating the URL group would be fine since you don't specify any URL prefixes during creation, you just create a new one which links it to the server's session. 

It's later on when you try to add a URL prefix to the previously created URL group that you get the error `ERROR_ALREADY_EXISTS`.

Specifically here:

https://github.com/dotnet/aspnetcore/blob/ffb8eb2cce47016b9436df1cf796968cf6caddf5/src/Servers/HttpSys/src/NativeInterop/UrlGroup.cs#L116-L123

One odd thing is if you start the second server with a different URL prefix, with some changes I made locally, it will attach to the queue, create a new session, a new URL group, and register the new prefix with its URL group. The odd part is if you make a request to the new URL, the first server actually receives the request. I didn't expect that. I'll ask the http.sys team about it.

**Comment by NGloreous** [comment]
I checked back in with the http.sys team and their suggestion was to just aways try to create the URL prefixes when attaching and handle the error (or call HttpFindUrlGroupId before trying to create). 

I'm thinking one option would be to try and create the URL prefixes if one was explicitly specified? The other option would be to add another mode `CreateOrAttachAndCreateUrlPrefix` (or something like that). 

As for the odd behavior I saw with two servers attached to the same queue with different prefixes. They actually do both end up serving traffic to either URL prefix.
